src/irule.cpp

```diff
@@ -322,11 +325,25 @@ main( int argc, char **argv ) {
                 if ( rulegen ) {
                     if ( startsWith( buf, "INPUT" ) || startsWith( buf, "input" ) ) {
                         gotRule = 1;
-                        trimSpaces( trimPrefix( buf ) );
+                        constexpr auto skip_len{5};
+                        const auto copy_length{std::strlen(buf) + 1 - skip_len};
+
+                        // Blank out "INPUT"
+                        std::memmove(buf, buf + skip_len, copy_length);
+
+                        auto res{boost::algorithm::trim_left_copy(std::string{buf})};
+                        std::strncpy(buf, res.c_str(), res.size() + 1);
                     }
                     else if ( startsWith( buf, "OUTPUT" ) || startsWith( buf, "output" ) ) {
                         gotRule = 2;
-                        trimSpaces( trimPrefix( buf ) );
+                        constexpr auto skip_len{6};
+                        const auto copy_length{std::strlen(buf) + 1 - skip_len};
```

Comment:
Can be `constexpr` too.

Reply:
Ignore this.

See conversation starting at https://github.com/irods/irods_client_icommands/pull/487#discussion_r1645185339.